### DISCUSSION TITLE
Workaround for the sidebar config using `_meta.json` when folders and files are prefixed with numbers

### DISCUSSION THREAD

**Discussion by XavierLeTohic** [web-infra-dev/rspress#1718] [Q&A] [answered]
Hi, a few days ago Rspress used to try to read directories as files if they were prefix with digits and dot like `05.onboarding/` before it was fixed (big kudos to @SoonIter btw, very appreciated!).

Now I'm facing the fact that when directories and/or files are prefixed with digits and dots (for versioning convention) I need to explicitly set the filename in the `_meta.json` with the digits and dot to get the siderbar showing and I have to set a custom display name so I don't show the digits and dot.

My team isn't willing to migrate to Rspress unless they need the least effort to make when updating the digits on each markdown changed.

Any workaround you would have in mind? A kind of regex or anything? Thank you very much!

**Comment by SoonIter** [comment, marked as answer]
you can use glob, and set https://rspress.dev/api/config/config-theme#sidebar manually and programmatically

```ts
import { defineConfig } from 'rspress/config';
import glob from 'fast-glob';

const files = glob.sync('./docs/foo/**/*.{md,mdx}');

const sidebars = files.map(() => {
	return [...]
})

export default defineConfig({
  themeConfig: {
    sidebar: {
      '/guide/': [
        {
          text: 'Getting Started',
          items: [
            // Fill in an object
            {
              text: 'Introduction',
              link: '/guide/getting-started/introduction',
            },
            {
              text: 'Installation',
              link: '/guide/getting-started/installation',
            },
            ...sidebars
          ],
        },
  
      ],
    },
  },
});
```

**Reply by XavierLeTohic** [reply]
Thank you, ended up generating the whole sidebar as you suggest, I was hoping for some king of native support like mkdocs does.
